lib/zendesk_api/middleware/response/sanitise_response.rb

```diff
@@ -0,0 +1,13 @@
+module ZendeskAPI
+  module Middleware
+    module Response
+      class SanitiseResponse < Faraday::Response::Middleware
+        def on_complete(env)
+          unless env[:body].strip.empty?
+            env[:body] = env[:body].encode('UTF-8', 'binary', invalid: :replace, undef: :replace, replace: '')
```

Comment:
@grosser I tried that before, but it was not working for 2.0.0. Nonetheless what I had done was wrong since the symbols such as ```€``` were being scrubbed as well.

@ggrossman I tried to use the credit card example but it did not work for JRuby. There was a problem with the null character and JSON.parse.

I ended up with a solution that is similar to the credit_card solution, however instead of replacing with the null character, I basically remove the character. This will avoid having different responses for the MRI and for JRuby

Somehow the tests are hanging in the infinite loop test in this jruby-1.7.19  + jdk 1.7.0_71. It works fine with JDK version 1.8+. Any idea how to solve this?